src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -2130,6 +2128,12 @@ public static bool IsExpectedException(Exception e)
             return e is OperationCanceledException ||
                    e is ObjectDisposedException;
         }
+
+
+        internal static void DisconnectAction(Task<RachisConnection> connectionTask)
+        {
```

Comment:
Maybe assert here that the `connectionTask` is completed?

Reply:
done.
But it can be faulted, so I only assert for `IsCompleted`.